src/msw/notebook.cpp

```diff
@@ -729,22 +730,195 @@ bool wxNotebook::InsertPage(size_t nPage,
+    // for each tab on the x axis.
+    wxDirection tabDir = GetTabOrientation();
+    if (!(tabDir == wxLEFT || tabDir == wxRIGHT))
+        return false;
+
+    long dummy = 0;
+    if (!flags)
+        flags = &dummy;
+
+    wxRect r = GetRect();
+    if (!r.Contains(pt))
+    {
+        *flags = wxBK_HITTEST_NOWHERE;
+        return true;
+    }
+
+    size_t pages = GetPageCount();
+    if (!pages)
+    {
+        *flags |= wxBK_HITTEST_NOWHERE | wxBK_HITTEST_ONPAGE;
+        return true;
+    }
+
+    if (GetPageSize().Contains(pt))
+    {
+        *flags = wxBK_HITTEST_ONPAGE;
+        return true;
+    }
+
+    r = GetTabRect(item);
+    if (!r.Contains(pt))
+    {
+        *flags = wxBK_HITTEST_NOWHERE;
+        return true;
+    }
+
+    // Just a magic number between the icon and the start of the text
+    // I checked with different resolutions and it seems to be about it. Have to check with
+    // different scalings as well.
+    int gap = 8;
+
+    TCHAR buffer[256];
+    TCITEM ti;
+    ti.mask = TCIF_IMAGE|TCIF_TEXT;
+    ti.cchTextMax = sizeof(buffer) - 1;
+    ti.pszText = buffer;
+    TabCtrl_GetItem(GetHwnd(), item, &ti);
+
+    wxRect tr;
+    wxRect ir;
+
+    // If the tab has an image, we need to check if the mouse is over it.
+    if (ti.iImage != -1)
+    {
+        wxIcon const &icon = GetImage(ti.iImage);
+        if (!icon.IsNull())
+        {
+            wxSize isz = icon.GetSize();
+            ir.width = isz.x;
+            ir.height = isz.y;
+        }
+    }
+
+    // The label text and the icon (if available) is centered in the tab, so we need
+    // to find out the size of that.
+    wxString label = ti.pszText;
+    const wxFont &labelFont = GetFont();
+    int len = label.length();
+    if (len)
+    {
+        // This is intentional and not a bug. :) On left/right the font is rotated 90 degree
+        // so we have to account for that by flipping width with height.
+        GetTextExtent(label, &tr.height, &tr.width, nullptr, nullptr, &labelFont);
+    }
+
+    // The tab has neither a text nor an icon
+    if (len == 0 && ir.width == 0)
+    {
+        *flags = wxBK_HITTEST_ONITEM;
+        return true;
+    }
+
+    if (len)
+    {
+        tr.x = r.x + (r.width - tr.width)/2;
+        if (tabDir == wxLEFT)
+        {
+            if (ir.height)
+                tr.y = r.y + (r.height - tr.height - gap - ir.height) / 2;
+            else
+                tr.y = r.y + (r.height - tr.height - ir.height) / 2;
+        }
+        else
+        {
+            if (ir.height)
+                tr.y = r.y + (r.height - tr.height + gap + ir.height) / 2;
+            else
+                tr.y = r.y + (r.height - tr.height) / 2;
+        }
+
+        if (tr.Contains(pt))
+        {
+            *flags = wxBK_HITTEST_ONLABEL;
+            return true;
+        }
+    }
+
+    if (ir.height)
+    {
+        wxDisplay disp(wxDisplay::GetFromWindow(this));
+        double scale = disp.GetScaleFactor();
```

Comment:
Couldn't we just do
```suggestion
        double scale = GetDPIScaleFactor();
```
and avoid using `wxDisplay`?